mypy/meet.py

```diff
@@ -753,13 +758,7 @@ def visit_instance(self, t: Instance) -> ProperType:
             if is_subtype(self.s.fallback, t):
                 return self.s
             return self.default(self.s)
-        elif isinstance(self.s, TypeType):
-            return meet_types(t, self.s)
-        elif isinstance(self.s, TupleType):
-            return meet_types(t, self.s)
-        elif isinstance(self.s, LiteralType):
-            return meet_types(t, self.s)
-        elif isinstance(self.s, TypedDictType):
+        elif isinstance(self.s, (TypeType, TupleType, LiteralType, TypedDictType, TypeVarType)):
```

Comment:
Note to self: check every visit method to make sure typevars are getting passed through.